Unit test TestOverlayCreateSnap fails (on Fedora)

This test is consistently failing on my host with:

```
--- FAIL: TestOverlayCreateSnap (0.00s)
        graphtest_unix.go:122: stat /var/tmp/docker-graphtest-723995303/overlay/Snap/merged/a subdir: no such file or directory
FAIL
coverage: 31.9% of statements
```

docker info
```
Containers: 1
 Running: 1
 Paused: 0
 Stopped: 0
Images: 167
Server Version: 1.11.0-dev
Storage Driver: overlay
 Backing Filesystem: extfs
Logging Driver: journald
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.4.6-300.fc23.x86_64
Operating System: Fedora 23 (Workstation Edition)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 11.44 GiB
Name: localhost.localdomain
ID: JQDM:ZDU3:BYSR:SIMJ:QGDH:GBGW:MF7Q:VYLQ:UYFD:W5KY:H4WG:PIMF
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 17
 Goroutines: 48
 System Time: 2016-03-31T13:47:54.765083779+02:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: true
```

docker version
```
Client:
 Version:      1.11.0-dev
 API version:  1.24
 Go version:   go1.5.3
 Git commit:   47fa54a
 Built:        Thu Mar 31 13:44:02 2016
 OS/Arch:      linux/amd64
 Experimental: true

Server:
 Version:      1.11.0-dev
 API version:  1.24
 Go version:   go1.5.3
 Git commit:   47fa54a
 Built:        Thu Mar 31 13:44:02 2016
 OS/Arch:      linux/amd64
 Experimental: true
```
I don't have any clue about this - might be kernel version (?), this fails on 4.2.x kernel also
ping @LK4D4 @estesp @calavera @icecrime

--- a/daemon/graphdriver/graphtest/graphtest_unix.go
+++ b/daemon/graphdriver/graphtest/graphtest_unix.go
@@ -67,7 +67,7 @@ func getBaseLoopStats() (*syscall.Stat_t, error) {
 }
 
 func newDriver(t *testing.T, name string) *Driver {
-	root, err := ioutil.TempDir("/var/tmp", "docker-graphtest-")
+	root, err := ioutil.TempDir("", "docker-graphtest-")
 	if err != nil {
 		t.Fatal(err)
 	}

--- a/daemon/graphdriver/overlay/overlay.go
+++ b/daemon/graphdriver/overlay/overlay.go
@@ -120,7 +120,7 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		backingFs = fsName
 	}
 
-	// check if they are running over btrfs or aufs
+	// check if they are running over btrfs, aufs, zfs or overlay
 	switch fsMagic {
 	case graphdriver.FsMagicBtrfs:
 		logrus.Error("'overlay' is not supported over btrfs.")
@@ -130,6 +130,9 @@ func Init(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (grap
 		return nil, graphdriver.ErrIncompatibleFS
 	case graphdriver.FsMagicZfs:
 		logrus.Error("'overlay' is not supported over zfs.")
+		return nil, graphdriver.ErrIncompatibleFS
+	case graphdriver.FsMagicOverlay:
+		logrus.Error("'overlay' is not supported over overlay.")
 		return nil, graphdriver.ErrIncompatibleFS
 	}
 

